lib/web_ui/dev/test_runner.dart

```diff
@@ -850,3 +876,23 @@ class TestBuildInput {
 }
 
 class TestFailureException implements Exception {}
+
+/// Prints the name of the test, but suppresses all other test output.
+///
+/// This is useful to prevent pollution of logs by tests that are expected to
+/// fail.
+class NameOnlyReporter implements hack.Reporter {
+  NameOnlyReporter(hack.Engine testEngine) {
+    testEngine.onTestStarted.listen(_printTestName);
+  }
+
+  void _printTestName(hack.LiveTest test) {
+    print('Running ${test.groups.map((hack.Group group) => group.name).join(' ')} ${test.individualName}');
+  }
+
+  @override
+  void pause() {}
+
+  @override
+  void resume() {}
```

Comment:
We should respect `pause` and `resume`.
```suggestion
  bool _isPaused = false;

  void _printTestName(hack.LiveTest test) {
    if (_isPaused) {
      return;
    }
    print('Running ${test.groups.map((hack.Group group) => group.name).join(' ')} ${test.individualName}');
  }

  @override
  void pause() {
    _isPaused = true;
  }

  @override
  void resume() {
    _isPaused = false;
  }
```

Reply:
This reporter is effectively always paused by never printing anything, so I think that would be superfluous. I believe test name is printed before the test is paused. But if you think this is useful, I'll take the suggestion.